probing/README.md

```diff
@@ -1,4 +1,81 @@
+  edges-spr2_test.json      # test set predictions, in edge probing JSON format
+  log.log                   # training and eval log file (human-readable text)
+  model_state_eval_best.th  # PyTorch saved checkpoint
+  params.conf               # serialized parameter list
+```
+`jiant` uses [tensorboardX](https://github.com/lanpa/tensorboardX) to record loss curves and a few other metrics during training. You can view with:
+```
+tensorboard --logdir $JIANT_PROJECT_PREFIX/ep_cove_demo/run/tensorboard
+```
+
+You can use the `run/*_val.json` and `run/*_test.json` files to run scoring and analysis. There are some helper utilities which allow you to load and aggregate predictions across multiple runs. In particular:
+- [analysis.py](analysis.py) contains utilities to load predictions into a set 
+  of DataFrames, as well as to pretty-print edge probing examples.
+- [edgeprobe_preds_sandbox.ipynb](edgeprobe_preds_sandbox.ipynb) walks through 
+  some of the features in `analysis.py`
+- [analyze_runs.py](analyze_runs.py) is a helper script to process a set of 
+  predictions into a condensed `.tsv` format. It computes confusion matricies for each label and along various stratifiers (like span distance) so you can easily and quickly perform further aggregation and compute metrics like accuracy, precision, recall, and F1. In particular, the `run`, `task`, `label`, `stratifier` (optional), and `stratum_key` (optional) columns serve as identifiers, and the confusion matrix is stored in four columns: `tp_count`, `fp_count`, `tn_count`, and `tp_count`. If you want to aggregate over a group of labels (like SRL core roles), just sum the `*_count` columns for that group before computing metrics.
+- [edgeprobe_aggregate_analysis.ipynb](edgeprobe_aggregate_analysis.ipynb) is 
+  the notebook we used to generate the tables and figures in the paper. It's kind of messy, but ultimately just does some shallow processing over the output of `analyze_runs.py`. If you're trying to do your own analysis, the main idiosyncracies to know about are: for `ontonotes-coref`, use the `1` label instead of `_micro_avg_`, and for `srl-conll2012` we report a `_clean_micro_` metric which aggregates all the labels that don't start with `R-` or `C-`.
+
+## Running the experiments from the paper
+
+We provide a frozen branch, [`edgeprobe_frozen_feb2019`](https://github.com/jsalt18-sentence-repl/jiant/tree/edgeprobe_frozen_feb2019), which should reflect the master branch as of the final version of the paper.
+
+The configs in `jiant/config/edgeprobe_*.conf` are the starting point for the experiments in the paper, but are supplemented by a number of parameter overrides (the `-o` flag to `main.py`). We use a set of bash functions to keep track of these, which are maintained in [`jiant/scripts/edges/exp_fns.sh`](../scripts/edges/exp_fns.sh).
+
+To run a standard experiment, you can do something like:
+```sh
+export NOTIFY_EMAIL="yourname@gmail.com"  # TODO: make it work without this
```

Comment:
Do we want to leave this here, or make an issue?

Reply:
Fixed it so it's no longer needed.